babel/numbers.py

```diff
@@ -641,68 +698,134 @@ def __init__(self, pattern, prefix, suffix, grouping,
         self.frac_prec = frac_prec
         self.exp_prec = exp_prec
         self.exp_plus = exp_plus
-        if '%' in ''.join(self.prefix + self.suffix):
-            self.scale = 2
-        elif u'‰' in ''.join(self.prefix + self.suffix):
-            self.scale = 3
-        else:
-            self.scale = 0
 
     def __repr__(self):
         return '<%s %r>' % (type(self).__name__, self.pattern)
 
-    def apply(self, value, locale, currency=None, force_frac=None):
-        frac_prec = force_frac or self.frac_prec
+    @property
+    def scale(self):
```

Comment:
Turning a _precomputed_ field into a property is going to have a small performance impact.  My first motivation to contribute to this module was a use case we had, where currency formatting was consuming a lot of CPU time.  This is not Java, so we don't have a JIT that can inline this method call.

If you have time, try to do some trivial benchmarking by comparing against the parent commit.  Think about formatting hundreds of thousands of values.